ext/standard/string.c

```diff
@@ -1866,6 +1866,42 @@ PHP_FUNCTION(str_contains)
 }
 /* }}} */
 
+/* {{{ proto bool str_starts_with(string haystack, string needle)
+   Checks if haystack starts with needle */
+PHP_FUNCTION(str_starts_with) {
+    zend_string *haystack, *needle;
```

Comment:
I just noticed that you used 4-spaces indentation, but this project uses tabs for C files (cf https://github.com/php/php-src/blob/master/.editorconfig)